compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -41,6 +45,34 @@ impl MetaVarExpr {
         check_trailing_token(&mut tts, sess)?;
         let mut iter = args.trees();
         let rslt = match ident.as_str() {
+            "concat" => {
+                fn element<'sess>(
+                    iter: &mut RefTokenTreeCursor<'_>,
+                    sess: &'sess ParseSess,
+                    span: Span,
+                ) -> PResult<'sess, MetaVarExprConcatElem> {
+                    let is_var = try_eat_dollar(iter);
+                    let ident = parse_ident(iter, sess, span)?;
+                    Ok(MetaVarExprConcatElem { ident, is_var })
+                }
+
+                let mut rslt = Vec::new();
+                rslt.push(element(&mut iter, sess, ident.span)?);
+                if !try_eat_comma(&mut iter) {
+                    return Err(sess.dcx.struct_span_err(ident.span, "expected comma"));
+                }
+                rslt.push(element(&mut iter, sess, ident.span)?);
```

Comment:
Can this be pushed inside the loop? I think we can check `N >= 2` later?

Reply:
Done